sklearn/tree/_classes.py

```diff
@@ -97,6 +99,30 @@ class BaseDecisionTree(MultiOutputMixin, BaseEstimator, metaclass=ABCMeta):
     Use derived classes instead.
     """
 
+    _parameter_constraints = {
+        "splitter": [StrOptions({"best", "random"})],
+        "max_depth": [Interval(Integral, 1, None, closed="left"), None],
+        "min_samples_split": [
+            Interval(Integral, 2, None, closed="left"),
+            Interval(Real, 0.0, 1.0, closed="neither"),
```

Comment:
Hang on, after changing this the tests fail:
`pytest -vl sklearn/tests/test_common.py -k check_param_validation`

fails with `AssertionError: DecisionTreeRegressor does not raise an informative error message when the parameter min_samples_split does not have a valid type or value.`
maybe this is also because of the int/float overlap? Although neither 1.0 (as in 100%) nor 1 (as in one sample) would be a valid parameter for `min_samples_split` right?

Reply:
There's a bug in the common test. Let me fix that in a separate PR

Reply:
PR opened https://github.com/scikit-learn/scikit-learn/pull/23513. Let's wait for it to be merged